examples/basics/UI.py

```diff
@@ -0,0 +1,246 @@
+    trajectory_files = [] if trajectory_files is None else trajectory_files
+    for i in range(len(trajectory_files)):
+        # Create PlaybackSimulation from trajectory and state files
+        recording_playbacks.append(
+            PlaybackSimulation(
+                name="recording-playback_" + str(i),
+                traj=trajectory_files[i],
+                state=state_file[i],
+            )
+        )
+    # Create OmniRunner with basic server
+    imd_runner = OmniRunner.with_basic_server(
+        *tuple(simulation_files + recording_playbacks), name=server_name, port=port
+    )
+    imd_runner.next()
+    imd_runner.runner.play_step_interval = 1 / simulation_fps
+    if start_paused:
+        imd_runner.pause()
+    if record_trajectory or record_shared_state:
+        # Record from server if required
+        record_from_server(
+            f"localhost:{imd_runner.app_server.port}",
+            trajectory_output_file,
+            shared_state_file,
+        )
+
+    return f"Simulation started with type: {simulation_type}, settings: {locals()}"
+
+
+def stop_simulation():
+    global imd_runner
+    try:
+        # Attempt to close the simulation runner
+        if imd_runner is not None:
+            imd_runner.close()
+    except NameError as e:
+        return e
+    return "Simulation stopped!"
+
+
+def create_ui():
+    with gr.Blocks(theme=gr.themes.Soft(), fill_height=True) as demo:
+        gr.Markdown("# Nanover IMD Python Server GUI")
+
+        # Radio button to select simulation type
+        simulation_type = gr.Radio(
+            ["From xml", "From recording"], label="Simulation Type", value="From xml"
+        )
+
+        with gr.Row():
+            with gr.Column(visible=True) as realtime_col:
+                # File input for live simulation
+                input_files = gr.File(
+                    label="Input Files (for From xml)", file_count="multiple"
+                )
+            with gr.Column(visible=False) as playback_col:
+                # File inputs for playback simulation
+                trajectory_files = gr.File(
+                    label="Trajectory Files (for playback)", file_count="multiple"
+                )
+                state_file = gr.File(
+                    label="State File (for playback)", file_count="multiple"
+                )
+
+        with gr.Row():
+            with gr.Column():
+                gr.Markdown("## Verbosity")
+                # Radio button for verbosity level
+                verbosity = gr.Radio(
+                    ["Normal", "Verbose", "Super verbose"],
+                    label="Verbosity Level",
+                    value="Normal",
+                )
+                show_progression = gr.Checkbox(label="Show simulation progression")
+
+                gr.Markdown("## Network")
+                # Textbox for server name
+                server_name = gr.Textbox(
+                    label="Server name", value="NanoVer-RS iMD Server"
```

Comment:
The name "NanoVer-RS" refers to the Rust server. This should be changed, perhaps to simply "NanoVer"? This also applies to the automatic file names for the output trajectory and state files.

Another thing is that if this name is being used to name files by default (does the rust UI do this?), we may wish to swap out the spaces for either underscores or hyphens, so that the names are easier to type in the command line!